### PR TITLE
Add hidden flag to disable permissions backup

### PR DESCRIPTION
**Pull request by meain**
## Description

This will let users completely disable backing up permissions altogether.

## Does this PR need a docs update or release note?

- [x] :white_check_mark: Yes, it's included
- [ ] :clock1: Yes, but in a later PR
- [ ] :no_entry: No 

## Type of change

- [x] :sunflower: Feature
- [ ] :bug: Bugfix
- [ ] :world_map: Documentation
- [ ] :robot: Test
- [ ] :computer: CI/Deployment
- [ ] :broom: Tech Debt/Cleanup

## Issue(s)

* fixes https://github.com/alcionai/corso/issues/2252

## Test Plan

- [ ] :muscle: Manual
- [x] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -18,6 +18,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - OneDrive file/folder permissions can now be backed up and restored
 - Add `--restore-permissions` flag to toggle restoration of OneDrive permissions
+- Add a hidden `--disable-permissions-backup` to completely disable backing up permissions
```
Comment (ryanfkeepers) [inline CHANGELOG.md:21]: Do we want to announce hidden flags in the changelog? 🤔

File: src/cli/options/options.go [hunk 2]
```diff
@@ -78,3 +82,16 @@ func DisableIncrementals() func(*pflag.FlagSet) {
...
+
+// Adds the hidden '--disable-permissions-backup' cli flag which, when
+// set, disables backing up permissions.
+func DisablePermissionsBackup() func(*pflag.FlagSet) {
+	return func(fs *pflag.FlagSet) {
+		fs.BoolVar(
+			&disablePermissionsBackup,
+			"disable-permissions-backup",
+			false,
+			"Disable backing up permissions for OneDrive")
```
Comment (ryanfkeepers) [inline src/cli/options/options.go:94]:
```suggestion
			"Disable backing up item permissions for OneDrive")
```

File: src/pkg/control/options.go [hunk 3]
```diff
@@ -75,4 +75,10 @@ type Toggles struct {
 	// DisableIncrementals prevents backups from using incremental lookups,
 	// forcing a new, complete backup of all data regardless of prior state.
 	DisableIncrementals bool `json:"exchangeIncrementals,omitempty"`
+
+	// DisablePermissionsBackup is used to disable backups for
+	// permissions. This can be used by other tools to completely
+	// disable the backup and restore functionality for backups as
+	// they consume more API credits and can cause increased throttling.
```
Comment (ryanfkeepers) [inline src/pkg/control/options.go:82]:
```suggestion
	// DisablePermissionsBackup is used to disable backups of item
	// permissions. Permission metadata increases graph api call count,
	// so disabling their retrieval when not needed is advised.
```

**Review by ryanfkeepers — APPROVED** [review]
Some nits, but looks good overall

File: src/internal/connector/onedrive/collection.go [hunk 4]
```diff
@@ -258,9 +267,38 @@ func (oc *Collection) populateItems(ctx context.Context) {
...
+				if !oc.ctrl.ToggleFeatures.DisablePermissionsBackup {
+					itemMeta, itemMetaSize, err = oc.itemMetaReader(ctx, oc.service, oc.driveID, item)
+				} else {
+					// We are still writing the metadata file but with
+					// empty permissions as we are not sure how the
+					// restore will be called.
+					itemMeta = io.NopCloser(strings.NewReader("{}"))
+					itemMetaSize = 2
+					err = nil
+				}
```
Comment (ashmrtn) [inline src/internal/connector/onedrive/collection.go:301]: it's a little hard to follow when we set err = nil so we hit the break in the else-block. It would be more straightforward to read if we just make an if-block that breaks when needed and let everything else have it's own control flow
```suggestion
				if oc.ctrl.ToggleFeatures.DisablePermissionsBackup {
					// We are still writing the metadata file but with
					// empty permissions as we are not sure how the
					// restore will be called.
					itemMeta = io.NopCloser(strings.NewReader("{}"))
					itemMetaSize = 2

					break
				}

				itemMeta, itemMetaSize, err = oc.itemMetaReader(ctx, oc.service, oc.driveID, item)
```

File: src/internal/connector/onedrive/collection.go [hunk 5]
```diff
@@ -258,9 +267,38 @@ func (oc *Collection) populateItems(ctx context.Context) {
...
+			// TODO(meain): refactor these multiple retry logic
 			for i := 1; i <= maxRetries; i++ {
-				itemInfo, itemData, itemMeta, err = oc.itemReader(ctx, oc.service, oc.driveID, item)
+				if !oc.ctrl.ToggleFeatures.DisablePermissionsBackup {
+					itemMeta, itemMetaSize, err = oc.itemMetaReader(ctx, oc.service, oc.driveID, item)
+				} else {
+					// We are still writing the metadata file but with
+					// empty permissions as we are not sure how the
+					// restore will be called.
+					itemMeta = io.NopCloser(strings.NewReader("{}"))
```
Comment (ashmrtn) [inline src/internal/connector/onedrive/collection.go:298]: given that we're deciding to store "empty" permissions and we now have flags for restoring and backing up permissions it feels like we should be printing a warning or some other message if a user tries to do `restore permissions(backup no permissions)`

This should be done in a separate PR if we decide to add it, but wanted to surface it here for discussion

Comment (meain) [reply src/internal/connector/onedrive/collection.go:298]: I guess that would be useful. I had a similar question in https://github.com/alcionai/corso/issues/2230#issuecomment-1402331725 . The problem with warning when we backup with `--disable-permissions-backup` is that we are not adding any data that as of now mentions that we avoided backing up permissions of it just didn't have any permissions.

Comment (ashmrtn) [reply src/internal/connector/onedrive/collection.go:298]: made #2357 to track this

File: src/internal/connector/onedrive/collection.go (hunk 4 above)
Comment (meain) [reply src/internal/connector/onedrive/collection.go:301]: Woops, this is such an obvious refactor :D

Comment (ashmrtn) [reply src/internal/connector/onedrive/collection.go]: no worries, it can be easy to miss stuff like this if there's a lot of other things going on too 😄
